src/cromshell/utilities/io_utils.py

```diff
@@ -101,6 +103,59 @@ def assert_path_is_not_empty(path: Union[str, Path], description: str) -> None:
             raise EOFError(f"ERROR: {description} is empty: {path}.")
 
 
+def has_nested_dependencies(wdl_path: str or Path) -> bool:
+    """Determine if a WDL has any nested imports."""
+
+    with open(wdl_path, 'r') as rf:
+        for l in rf:
+            if l.startswith('import'):
+                m = re.match(r'import "(.+)"', l)
+
+                if "../" in m.group(1):
+                    return True
+
+    return False
+
+
+def get_flattened_filename(tempdir: tempfile.TemporaryDirectory, wdl_path: str or Path):
+    """Generate the filename to use for flattened WDL files."""
+
+    return os.path.join(
+        tempdir.name,
+        re.sub("^-", "", re.sub(os.path.sep, "-", os.path.dirname(wdl_path))) + '-' + os.path.basename(wdl_path)
+    )
+
+
+def flatten_nested_dependencies(tempdir: tempfile.TemporaryDirectory, wdl_path: str or Path) -> str:
+    """Flatten a WDL directory structure and rewrite imports accordingly.
+
+    Return string representing the filesystem location of the rewritten WDL.
+    """
+
+    wdl_dir = os.path.dirname(wdl_path)
+
+    new_wdl_path = get_flattened_filename(tempdir, wdl_path)
+
+    with open(wdl_path, 'r') as rf, open(new_wdl_path, 'w') as wf:
+        for l in rf:
+            if l.startswith('import'):
+                m = re.match(r'import "(.+)"', l)
+
```

Comment:
Not needed but to improve readability it would be helpful to put m.group(1) into a variable and use it in the lines below. 
```suggestion
                imported_wdl_name = m.group(1)
```